src/sushi.sol

```diff
@@ -186,19 +198,24 @@ contract SushiJoin is CropJoin {
         );
         require(
             selector == MasterChefLike.transferOwnership.selector ||
-            selector == MasterChefLike.setMigrator.selector
+            selector == MasterChefLike.setMigrator.selector ||
+            selector == MasterChefLike.set.selector
         , "SushiJoin/wrong-function");
+        if (selector == MasterChefLike.set.selector) {
+            uint8 overwrite = uint8(bytes(signature).length == 0 ? data[131] : data[127]) & 0x1;
```

Comment:
Could we avoid the conditional here by reading byte 131 of the `callData` variable?

Reply:
Also, more generally--is it really necessary to support passing an empty `signature`, since it seems that in that case the caller just has to include the selector in `data`? I personally don't see the need to support two ways of doing the same thing, and it makes the code more complex. Personally I'd remove the `signature` arg, rename `data` to `callData`, and then expect the first four bytes to be the selector (with appropriate checks).

Reply:
Updated the `data` to be `callData`.  The double way to specify the signature and calldata is kind of dumb I agree, but I have to copy what is in the timelock contract. Removing one or the other would allow a valid signature to be specified in the timelock for which we could not cage on as `bytes32 txHash = keccak256(abi.encode(masterchef, value, signature, data, eta));` would change depending on the style used.

Reply:
Ah yeah I see, we have no choice in the matter, the stupidity is forced upon us :upside_down_face: .